velox/expression/ExprConstants.h

```diff
@@ -15,16 +15,67 @@
  */
 #pragma once
 
+/// This file contains string literals for function names.
+/// These literals listed in single place to explicitly list functions that
+/// names used once even in case when user specify some prefix on registration.
+
 namespace facebook::velox::expression {
 
-constexpr const char* kAnd = "and";
-constexpr const char* kOr = "or";
-constexpr const char* kSwitch = "switch";
-constexpr const char* kIf = "if";
-constexpr const char* kFail = "fail";
-constexpr const char* kCoalesce = "coalesce";
-constexpr const char* kCast = "cast";
-constexpr const char* kTryCast = "try_cast";
-constexpr const char* kTry = "try";
+/// These functions registered without prefix and velox code rely on this.
+inline constexpr const char* kAnd = "and";
+inline constexpr const char* kOr = "or";
+inline constexpr const char* kIn = "in";
+inline constexpr const char* kCast = "cast";
+inline constexpr const char* kTryCast = "try_cast";
+inline constexpr const char* kIsNull = "is_null";
+inline constexpr const char* kRowConstructor = "row_constructor";
+
+#ifdef VELOX_ENABLE_BACKWARD_COMPATIBILITY
+inline constexpr const char* kSwitch = "switch";
+inline constexpr const char* kIf = "if";
+inline constexpr const char* kFail = "fail";
+inline constexpr const char* kCoalesce = "coalesce";
+inline constexpr const char* kTry = "try";
+#endif
+
+/// These functions are also registered without prefix, but right now velox code
+/// doesn't rely on this. So please try to avoid such dependency in future,
+/// because it can be changed. To discourage usage of these literals in new code
+/// we put them to `old` namespace.
+namespace old {
+
+/// common, presto and spark special form functions without prefix
+inline constexpr const char* kCoalesce = "coalesce";
+inline constexpr const char* kIf = "if";
+inline constexpr const char* kSwitch = "switch";
+inline constexpr const char* kTry = "try";
+
+/// spark special form functions without prefix
+inline constexpr const char* kMakeDecimal = "make_decimal";
+inline constexpr const char* kRoundDecimal = "decimal_round";
+inline constexpr const char* kAtLeastNNonNulls = "at_least_n_non_nulls";
+inline constexpr const char* kFromJson = "from_json";
+inline constexpr const char* kGetStructField = "get_struct_field";
+inline constexpr const char* kGetArrayStructFields = "get_array_struct_fields";
+inline constexpr const char* kConcatWs = "concat_ws";
```

Comment:
These seem to be specific to Spark. How about moving them under sparksql to avoid exposing them in the common library? Thanks.

Reply:
In such case more easy to just avoid this PR (I made same PR but without spark), the whole point was to have list names that doesn't have prefixes so cannot be easily used even for different runtime.